src/os/bluestore/BlueStore.cc

```diff
@@ -6452,11 +6454,16 @@ int BlueStore::_open_db(bool create, bool to_repair_db, bool read_only)
   string kv_dir_fn;
   string kv_backend;
   std::string sharding_def;
+  // prevent write attempts to BlueFS in case we failed before BlueFS was opened
+  db_was_opened_read_only = true;
   r = _prepare_db_environment(create, read_only, &kv_dir_fn, &kv_backend);
   if (r < 0) {
     derr << __func__ << " failed to prepare db environment: " << err.str() << dendl;
     return -EIO;
   }
+  // if reached here then BlueFS is already opened
+  db_was_opened_read_only = read_only;
+  dout(5) << __func__ << "::NCB::db_was_opened_read_only was set to " << read_only << dendl;
```

Comment:
suggest to raise the debug level to 10 and indicate ro vs. rw mode irrespective to NCB.